odml/tools/rdf_converter.py

```diff
@@ -0,0 +1,141 @@
+        if docs:
+            self.g.add((self.hub_root, odmlns.Hub, URIRef(odmlns.Hub)))
+            for doc in docs:
+                self.save_element(doc)
+        return self.g
+
+    def save_element(self, e, node=None):
+        """
+        Save the current element to the RDF graph
+        :param e: current element 
+        :param node: Blank node to pass the earlier created node to inner elements
+        :return: the RDF graph 
+        """
+        fmt = e._format
+
+        if not node:
+            curr_node = URIRef(odmlns + str(e.id))
+        else:
+            curr_node = node
+
+        self.g.add((curr_node, fmt.rdf_type(), URIRef(fmt.rdf_type())))
+
+        # adding doc to the hub
+        if isinstance(fmt, format.Document.__class__):
+            self.g.add((self.hub_root, odmlns.hasDocument, curr_node))
+
+        for k in fmt._rdf_map:
+            if k == 'id':
+                continue
+            elif (isinstance(fmt, format.Document.__class__) or
+                      isinstance(fmt, format.Section.__class__)) and k == "repository":
+                terminology_url = getattr(e, k)
+                if terminology_url is None or not terminology_url:
+                    continue
+                terminology_node = self._get_terminology_by_value(terminology_url)
+                if terminology_node:
+                    self.g.add((curr_node, fmt.rdf_map(k), terminology_node))
+                else:
+                    # adding terminology to the hub and to link with the doc
+                    node = URIRef(odmlns + str(uuid.uuid4()))
+                    self.g.add((node, odmlns.Terminology, URIRef(terminology_url)))
+                    self.g.add((self.hub_root, odmlns.hasTerminology, node))
+                    self.g.add((curr_node, fmt.rdf_map(k), node))
+            # generating nodes for entities: sections, properties and bags of values
+            elif (isinstance(fmt, format.Document.__class__) or
+                      isinstance(fmt, format.Section.__class__)) and \
+                            k == 'sections' and len(getattr(e, k)) > 0:
+                sections = getattr(e, k)
+                for s in sections:
+                    node = URIRef(odmlns + str(s.id))
+                    self.g.add((curr_node, fmt.rdf_map(k), node))
+                    self.save_element(s, node)
+            elif isinstance(fmt, format.Section.__class__) and \
+                            k == 'properties' and len(getattr(e, k)) > 0:
+                properties = getattr(e, k)
+                for p in properties:
+                    node = URIRef(odmlns + str(p.id))
+                    self.g.add((curr_node, fmt.rdf_map(k), node))
+                    self.save_element(p, node)
+            elif isinstance(fmt, format.Property.__class__) and \
+                            k == 'value' and len(getattr(e, k)) > 0:
+                values = getattr(e, k)
+                bag = BNode(str(uuid.uuid4()))
```

Comment:
We should either use a proper blank node without UUID or use a proper URIref with a UUID for values here and add `self.g.add(bag, RDF.type, RDF.Bag)` in the next line.
I would be in favor of not using Blank Nodes until we get the chance to read up on the topic, since Blank Nodes can cause problems in SPARQL queries later on.